### PR TITLE
Add missing imports for documentation builds

### LINKED ISSUE #3075: docs building is missing import
Since we added the architecture slides that used recommonmark jenkins is failing probably because we do not have the wheel to satisfy `from recommonmark.parser import CommonMarkParser`

https://jenkins.galaxyproject.org/job/latest-Sphinx-Docs/lastFailedBuild/console

recommonmark added here:
https://github.com/galaxyproject/galaxy/commit/3e737c99ea21dcb7ea3a326d36aee9c715c8d70b#diff-555c472a89566734e0d8567fb205d5dfR19

### PR DESCRIPTION
**Pull request by athos-ribeiro**
As mentioned in #3075, the build requirements for the project
documentation were only listed in the Jenkins job. This commit adds
those missing requirements to the project. Note that they should also be
removed from the Jenkins build
- fixes #3075

### CODE REVIEW

**Comment by jmchilton** [comment]
Thanks for the fix! These aren't required runtime by Galaxy - so these should be placed into dev-requirements in that same directory instead. Unfortunately we build binary dependencies for all of our requirements so more work is needed before even that change can be merged. I created an issue for this here https://github.com/galaxyproject/galaxy/issues/3166.

I'm going to mark this as WIP until this issue is resolved and I opened a PR to at least document this implicit requirement (https://github.com/galaxyproject/galaxy/pull/3167). Thanks again!

**Comment by bgruening** [comment]
@athos-ribeiro can you address @jmchilton comments? Would be nice to get this fixed :) Thanks!

**Comment by bgruening** [comment]
ping @athos-ribeiro

**Comment by martenson** [comment]
This is now ready for review. @bgruening @jmchilton 

(the wheels are not necessary for this)

File: Makefile [hunk 1]
```diff
@@ -22,14 +22,15 @@ OPEN_RESOURCE=bash -c 'open $$0 || xdg-open $$0'
...
 
-# Building docs requires sphinx and utilities be installed (see issue 3166) as well as pandoc.
-# Run following commands to setup the Python portion of these requirements:
+docs:
+# Generate HTML documentation.
+# Run following commands to setup the Python portion of the requirements:
 #   $ ./scripts/common_startup.sh
 #   $ . .venv/bin/activate
-#   $ pip install sphinx sphinx_rtd_theme lxml recommonmark
-docs: ## generate Sphinx HTML documentation, including API docs
```
Comment (nsoranzo) [inline Makefile:32]: This change will break the `help` target, which requires the syntax:
```
target: ## target help
```

Comment (martenson) [reply Makefile:32]: Thanks! I think I fixed this below.

**Comment by nsoranzo** [comment]
`py27-unit` and `qunit` builds are broken by the `babel` requirement from `sphinx` which probably is conflicting with our `Babel` 2.0 wheel.

**Comment by martenson** [comment]
@nsoranzo should we move this to different requirement file that does not get installed during tests? :/

**Comment by jmchilton** [comment]
@martenson We should update our babel I'd imagine. Let I'll open a PR that does that and see if the tests run fine?

**Comment by jmchilton** [comment]
~~Oh wait - nevermind... sphinx wants an older babel :(...~~

Update 2: Actually it just needs to be not exactly 2.0 - I'll try 2.4.

Update 3: xref https://github.com/sphinx-doc/sphinx/commit/6bce0a1c10f7ebeda7b54afed88c27dca185f1ee

**Comment by nsoranzo** [comment]
Updating Babel is a sensible idea, not sure if having a separate doc-requirements file is worthwhile.

**Comment by jmchilton** [comment]
xref #3986
xref https://github.com/galaxyproject/starforge/pull/128

**Comment by martenson** [comment]
thanks @jmchilton , I merged and restarted the build

**Comment by martenson** [comment]
yay! thanks @nsoranzo @jmchilton and @athos-ribeiro !

**Comment by jmchilton** [comment]
Fantastic - thanks for awesome contribution @athos-ribeiro and sorry for the delay.

**Pull request merged**